app/models/ext_management_system.rb

```diff
@@ -503,4 +503,23 @@ def blacklisted_event_names
   def self.blacklisted_events
     BlacklistedEvent.where(:provider_model => name, :ems_id => nil)
   end
+
+  def self.credential_validation(ems, auth_type, creds, revalidate)
```

Comment:
I agree.  If we need a class method, it can take an ems_id, look up the instance and call the instance method on that ems.